### ISSUE TITLE
The compiler should initialize arrays with a type

### ISSUE THREAD

**Issue by csmuller** [neow3j/neow3j#273]
When creating a new array, e.g., new boolean[10], the neow3j compiler uses the NEWARRAY neo-vm opcode. As a result the items in the array have the initial stack item type ANY which doesn't necessarily lead to errors but is definitely not optimal. Make the compiler use NEWARRAY_T with a type operand in cases where the type of the array is clear.

- Parse additional information on JVM's `NEWARRAY` and `ANEWARRAY` instructions to figure out the type of the array
- Initialize empty arrays with the neo-vm opcode `NEWARRAY0` because it is much cheaper.

**Comment by gsmachado** [comment]
@csmuller this can be seen as "optimization", right?

**Comment by csmuller** [comment]
It's not a optimization in the sense of performance or GAS costs. The problem is that an array initialized with NEWARRAY and no specific values will have values of type ANY. If you then go ahead and set one of the elements on the array to a specific value that element changes its type to the values type. When you return such an array the returned array will look something like this.
```
{ "type":"Array", 
   "value": [
        {"type":"Any"},
        {"type":"Boolean", "value":true}.
        {"type":"Any"},
        {"type":"Any"}
```

**Comment by csmuller** [comment]
Resolved by #438.
